packages/handlers/json-schema/src/index.ts

```diff
@@ -144,166 +54,10 @@ export default class JsonSchemaHandler implements MeshHandler {
-              default:
-                throw new Error(`Unknown method ${operationConfig.method}`);
-            }
-          }
-          operationLogger.debug(`=> Fetching ${urlObj.toString()}=>${inspect(requestInit)}`);
-          const response = await fetch(urlObj.toString(), requestInit);
-          const responseText = await response.text();
-          operationLogger.debug(
-            `=> Fetched from ${urlObj.toString()}=>{
-              body: ${responseText}
-            }`
-          );
-          const returnType = field.type;
-          let responseJson: any;
-          try {
-            responseJson = JSON.parse(responseText);
-          } catch (e) {
-            // The result might be defined as scalar
-            if (isScalarType(returnType)) {
-              operationLogger.debug(` => Return type is not a JSON so returning ${responseText}`);
-              return responseText;
-            }
-            throw responseText;
-          }
-          const errorMessageTemplate = this.config.errorMessage || 'message';
-          function normalizeError(error: any): Error {
-            const errorMessage = stringInterpolator.parse(errorMessageTemplate, error);
-            if (typeof error === 'object' && errorMessage) {
-              const errorObj = new Error(errorMessage);
-              errorObj.stack = null;
-              Object.assign(errorObj, error);
-              return errorObj;
-            } else {
-              return error;
-            }
-          }
-          const errors = responseJson.errors || responseJson._errors;
-          // Make sure the return type doesn't have a field `errors`
-          // so ignore auto error detection if the return type has that field
-          if (errors?.length) {
-            if (!('getFields' in returnType && 'errors' in returnType.getFields())) {
-              const aggregatedError = new AggregateError(
-                errors.map(normalizeError),
-                `${operationConfig.type}.${operationConfig.field} failed`
-              );
-              aggregatedError.stack = null;
-              this.logger.debug(`=> Throwing the error ${inspect(aggregatedError)}`);
-              return aggregatedError;
-            }
-          }
-          if (responseJson.error) {
-            if (!('getFields' in returnType && 'error' in returnType.getFields())) {
-              const normalizedError = normalizeError(responseJson.error);
-              operationLogger.debug(`=> Throwing the error ${inspect(normalizedError)}`);
-              return normalizedError;
-            }
-          }
-          operationLogger.debug(`=> Returning ${inspect(responseJson)}`);
-          return responseJson;
-        };
-      }
-    }
+    addExecutionLogicToComposer(schemaComposer, options);
 
-    this.logger.debug(`Building the executable schema.`);
-    const schema = schemaComposer.buildSchema();
     return {
-      schema,
+      schema: schemaComposer.buildSchema(),
```

Comment:
why do we do that extra logic here?  should we get the full GraphQL schema here already?
is it just because of the directives? if so, why isn't that also lives in `@omnigraph/json-schema`?

Reply:
Yes moved that part to the new package. Handler doesn't do anything about schemacomposers now.